tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -286,9 +288,22 @@ private void FeaturesProductConfigChanged(object sender, ProductConfigChangedEve
             e.Acknowledge(features.Name);
         }
 
+        private void AsmDataProductConfigChanged(object sender, ProductConfigChangedEventArgs e)
+        {
+            _asmDataConfigs ??= new Dictionary<string, Payload>();
+            var asmDataConfigs = e.GetDeserializedConfigurations<Payload>();
+            foreach (var asmDataConfig in asmDataConfigs)
+            {
+                _asmDataConfigs[asmDataConfig.Name] = asmDataConfig.TypedFile;
```

Comment:
I think you need to call `e.Acknowledge(asmDataConfig.Name);` here.

Reply:
fixed in f7f395e14e5bd4ff126f045189695a42c7fdbe72
I add an error if rules couldnt be updated instead of acklowledging